src/main/java/SaveDataHandler.java

```diff
@@ -0,0 +1,121 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Handles the saving and reading of tasks to/from a file.
+ */
+public class SaveDataHandler {
+
+    // Specifies relative path to save in the repo
+    private static final String SAVE_FILE_PATH = "data/Nyani.txt";
+
+    /**
+     * Saves the current task list to a file and then loads it.
+     *
+     * @param toDo the to-do list containing the tasks to save.
+     */
+    public static void saveData(todoList toDo) {
+
+        Path saveFilePath = Paths.get(SAVE_FILE_PATH);
+
+        try {
+            Files.createDirectories(saveFilePath.getParent());
+        } catch (IOException e) {
+            System.err.println("Could not create directory.");
+            return;
+        }
+
+        // Write tasks to specified file
+        writeSaveData(saveFilePath, toDo.getTaskStrings());
+
+        // Read saved tasks
+        readSaveData(toDo);
+
+    }
+
+    /**
+     * Writes the list of task strings to a file.
+     *
+     * @param path the path of the file to save the tasks to.
+     * @param tasks the list of task strings to save.
+     */
+    private static void writeSaveData(Path path, List<String> tasks) {
+        try {
+            Files.write(path, tasks);
+            System.out.println("Nyani will remember ur tasks for u!");
+        } catch (IOException e) {
+            System.err.println("Save failed :(");
+        }
+    }
+
+    /**
+     * Reads saved tasks from a file and adds them to the to-do list.
+     *
+     * @param toDo the to-do list to which the tasks will be added.
+     */
+    public static void readSaveData(todoList toDo) {
+        Path saveFilePath = Paths.get(SAVE_FILE_PATH);
+
+        if (Files.exists(saveFilePath)) {
+            try {
+                List<String> lines = Files.readAllLines(saveFilePath);
+                System.out.println("Loading tasks...");
+
+                for (String line : lines) {
+
+                    // Print each line being processed
+                    System.out.println(line);
+
+                    // Split the line into status and the rest of the info
+                    String[] parts = line.split("\\]", 2);
```

Comment:
consider moving low-level string operations to a different method for better SLAP